Assets/Scripts/ConnectionManagement/ConnectionState/HostingState.cs

```diff
@@ -71,9 +71,23 @@ public override void OnClientDisconnect(ulong clientId)
 
         public override void OnUserRequestedShutdown()
         {
-            m_ConnectionManager.SendServerToAllClientsSetDisconnectReason(ConnectStatus.HostEndedSession);
-            // Wait before shutting down to make sure clients receive that message before they are disconnected
-            m_ConnectionManager.StartCoroutine(WaitToShutdown());
+            ulong[] clientIds = new ulong[m_ConnectionManager.NetworkManager.ConnectedClientsIds.Count - 1];
+            var i = 0;
+            foreach (var id in m_ConnectionManager.NetworkManager.ConnectedClientsIds)
+            {
+                if (id != m_ConnectionManager.NetworkManager.LocalClientId)
+                {
+                    clientIds[i++] = id;
+                }
+            }
+
+            var reason = JsonUtility.ToJson(ConnectStatus.HostEndedSession);
+
+            foreach (var id in clientIds)
```

Comment:
why two for loops? Why not just do this in the first loop?

Reply:
Calling DisconnectClient removes clients from the loop which is not allowed while iterating. I suppose I could replace it with a single for(int i = Count - 1; i >= 0; i++) instead though if you think that it is cleaner

Reply:
i-- (not i++ :P )
but yeah, that'd make this a bit easier to understand.

Reply:
will do